traitsui/tests/_tools.py

```diff
@@ -99,6 +104,21 @@ def _is_current_backend(backend_name=""):
 is_mac_os = sys.platform == "Darwin"
 
 
+def requires_one_of(backends):
```

Comment:
When writing (non-null) toolkit-agnostic test case, `skip_if_null` still isn't quite future proof because the moment one adds a new toolkit, the tests will be failing with `NotImplementedError` on that toolkit, unless one captures and then reraise `SkipTest`. It is much easier to opt-in toolkits than to opt-out toolkits in this circumstance.

When the list of toolkits contains only one item, then the choice is somewhat subjective. To be honest, I'd still much prefer stating the positive `requires_one_of([WX])` or `requires(WX)` than the negative `skip_if_not(WX)`.  `requires_one_of` supports all these use cases I think. I don't mind keeping `skip_if_*` around, though we probably want only one way to do things if that's possible :).

Reply:
> Is this meant to eventually replace skip_if_*?

Sorry, so no that wasn't my intention, but I'd quite like for this `requires_one_of` to replace `skip_if_*`, if that's agreed by others :)